src/layers/recurrent.jl

```diff
@@ -636,3 +636,64 @@ function Base.show(io::IO, g::GRUCell{use_bias, TS}) where {use_bias, TS}
     TS && print(io, ", train_state=true")
     return print(io, ")")
 end
+
+"""
+    Bidirectional(cell::AbstractRecurrentCell;
+        backward_cell::Union{AbstractRecurrentCell, Nothing}=nothing,
+        merge_mode::Union{Function, Nothing}=vcat,
+        ordering::AbstractTimeSeriesDataBatchOrdering=BatchLastIndex())
+
+Bidirectional wrapper for RNNs.
+
+## Arguments
+
+  - `cell`: A recurrent cell. See [`RNNCell`](@ref), [`LSTMCell`](@ref), [`GRUCell`](@ref),
+    for how the inputs/outputs of a recurrent cell must be structured.
+
+## Keyword Arguments
+
+  - `backward_cell`: A optional backward recurrent cell. If `backward_cell` is `nothing`,
+    the rnn layer instance passed as the `cell` argument will be used to generate the
+    backward layer automatically. `in_dims` of `backward_cell` should be consistent with
+    `in_dims` of `cell`
+  - `merge_mode`: Function by which outputs of the forward and backward RNNs will be combined.
+    default value is `vcat`. If `nothing`, the outputs will not be combined.
+  - `ordering`: The ordering of the batch and time dimensions in the input. Defaults to
+    `BatchLastIndex()`. Alternatively can be set to `TimeLastIndex()`.
+
+## Inputs
+
+  - If `x` is a
+
+      + Tuple or Vector: Each element is fed to the `cell` sequentially.
+
+      + Array (except a Vector): It is spliced along the penultimate dimension and each
+        slice is fed to the `cell` sequentially.
+
+## Returns
+
+  - Merged output of the `cell` and `backward_cell` for the entire sequence.
+  - Update state of the `cell` and `backward_cell`.
+
+## Parameters
+
+  - Same as `cell` and `backward_cell`.
+
+## States
+
+  - Same as `cell` and `backward_cell`.
+"""
+function Bidirectional(cell::AbstractRecurrentCell;
+        backward_cell::Union{AbstractRecurrentCell, Nothing}=nothing,
+        merge_mode::Union{Function, Nothing}=vcat,
+        ordering::AbstractTimeSeriesDataBatchOrdering=BatchLastIndex())
+    if !isnothing(backward_cell) && cell.in_dims != backward_cell.in_dims
+        throw(DimensionMismatch(lazy"`in_dims` of `cell` and `backward_cell` should be the same"))
+    end
+    layer = Recurrence(cell; return_sequence=true, ordering)
+    backward_rnn_layer = isnothing(backward_cell) ? deepcopy(layer) :
+                         Recurrence(backward_cell; return_sequence=true, ordering)
+    fuse_op = isnothing(merge_mode) ? nothing : Broadcast.BroadcastFunction(merge_mode)
+    return Parallel(
```

Comment:
I was thinking maybe we create a struct

```julia
struct BidirectionalRNN <: ....
    model
end

(rnn::...)(...) = rnn.model(....)
```

This is mostly to be consistent with other wrapper layers in Lux which return the same type and not `Parallel` (as in this case)